lib/datadog/tracing/contrib/http/instrumentation.rb

```diff
@@ -35,6 +35,10 @@ def request(req, body = nil, &block)
                 span.type = Tracing::Metadata::Ext::HTTP::TYPE_OUTBOUND
                 span.resource = req.method
 
+                if Tracing::TraceOperation.appsec_standalone_reject?(trace)
+                  trace.sampling_priority = Tracing::Sampling::Ext::Priority::AUTO_REJECT
```

Comment:
This will override distributed tracing sampling priority, is that expected?

Reply:
Another thing, this likely has to be done for all places where `inject(trace, req)` happens.

Reply:
> Another thing, this likely has to be done for all places where inject(trace, req) happens.

If this is true, I suggest adding this logic to the shared `inject!` logic here: https://github.com/DataDog/dd-trace-rb/blob/7cd16a58f820845ce812877d32fa2a3129f9d291/lib/datadog/tracing/distributed/propagation.rb#L53

Reply:
Hmm, this logic here, from `lib/datadog/tracing/contrib/http/circuit_breaker.rb`, can probably also be moved to `inject!`:
```ruby
            if Datadog.configuration.appsec.standalone.enabled
              # Skip distributed tracing so that we don't bill distributed traces in case of absence of
              # upstream ASM event (_dd.p.appsec:1) and no local security event (which sets _dd.p.appsec:1 locally).
              # If there is an ASM event, we still have to check if distributed tracing is enabled or not
              return true unless Tracing.active_trace

              return true if Tracing.active_trace.get_tag(Datadog::AppSec::Ext::TAG_DISTRIBUTED_APPSEC_EVENT) != '1'
            end
```

Reply:
> Another thing, this likely has to be done for all places where `inject(trace, req)` happens.

I agree that this should most likely be done in all the places where there is `HTTP.inject`, however I don't think we should add it to `inject!`, as we must add `_sampling_priority_v1` to the trace even though it is not distributed. If we add it to inject and distributed tracing is disabled in the client config, it will not force the sampling priority to be rejected.

Reply:
> Another thing, this likely has to be done for all places where `inject(trace, req)` happens.

Fixed in e24310efa3ba270c6a90be6bd8e779b7d445a178